### DISCUSSION TITLE
[go/sdk] Conversion between pulumi built-in types and golang standard built-in types

### DISCUSSION THREAD

**Discussion by mritd** [pulumi/pulumi#9234] [Pulumi in Practice]
I understand the input and output of pulumi, but in some cases need to convert from pulumi output to golang standard built-in type;

For example, use Helm Chart's `GetResource` method, which accepts standard golang built-in type parameters, but uses pulumi's built-in types throughout the programming environment;

```go
// GetResource returns a resource defined by a built-in Kubernetes group/version/kind, name and namespace.
// For example, GetResource("v1/Pod", "foo", "") would return a Pod called "foo" from the "default" namespace.
func (c *Chart) GetResource(gvk, name, namespace string) pulumi.AnyOutput {
	id := name
	if len(namespace) > 0 && namespace != "default" {
		id = fmt.Sprintf("%s/%s", namespace, name)
	}
	key := fmt.Sprintf("%s::%s", gvk, id)
	return c.Resources.ApplyT(func(x interface{}) interface{} {
		resources := x.(map[string]pulumi.Resource)
		return resources[key]
	}).(pulumi.AnyOutput)
}
```

Is there some easy way to convert between pulumi built-in types and golang standard built-in types?

**Comment by Graham-Beer** [comment]
Would be nice to see less boiler plating to do the conversion. Would be interested to see if there is a better way to achieve this as well.
